godot-macros/src/class/derive_godot_class.rs

```diff
@@ -33,6 +33,23 @@ pub fn derive_godot_class(decl: Declaration) -> ParseResult<TokenStream> {
     let prv = quote! { ::godot::private };
     let godot_exports_impl = make_property_impl(class_name, &fields);
 
+    let editor_plugin = if struct_cfg.is_editor_plugin {
+        quote! {
+            ::godot::sys::plugin_add!(__GODOT_PLUGIN_REGISTRY in #prv; #prv::ClassPlugin {
+                class_name: #class_name_obj,
+                component: #prv::PluginComponent::EditorPlugin,
+                init_level: <#class_name as ::godot::obj::GodotClass>::INIT_LEVEL,
+            });
+
+            impl #class_name {
+                #[doc(hidden)]
+                const __INHERITS_EDITOR_PLUGIN: () = <#class_name as ::godot::obj::Inherits<::godot::engine::EditorPlugin>>::INHERITS;
+            }
```

Comment:
In [another place](https://github.com/godot-rust/gdext/blob/9e6fe56ab251529ea1d91d6df0ca978b3ad1e22a/godot-core/src/storage.rs#L213-L227), we have used a slightly different idiom to ensure a trait bound at compile time (although currently commented-out):

```rs
fn __static_type_check() {
    enforce_sync::<InstanceStorage<T>>();
}

fn enforce_sync<T: Sync>() {}
```
Do you see any advantages of one vs. the other idiom?

Maybe it makes sense to do generalize this to something like `static_assert_bounds!(Type : Bounds);` or so? Although a macro might just put more load on the codegen 🤔

Reply:
I don't really think there's a big difference either way. Though this one is more discoverable at least. I couldn't quite find a place this had been done before. I can change to doing it that way.